tests/test_tic_tac_toe.py

```diff
@@ -14,9 +15,25 @@ def get_input(self, message=""):
     def print_output(self, output):
         self.last_output = output
 
+@pytest.fixture
+def filled_board_output():
+    return ' O | X | O \n-----------\n X | O | X \n-----------\n O | X | O '
 
-def test_can_play_a_full_game():
-    mock_console_io = MockConsoleIO()
-    console = Console(console_io=mock_console_io)
+
+def test_can_play_a_full_game(filled_board_output):
```

Comment:
Someone one day showed me a pattern of writing tests that I try to follow as it makes the tests a bit faster to read. It's about splitting a text into 3 parts: preparation, action, assertion. In your case it would be:

```
def test_can_play_a_full_game(filled_board_output):
    values = ['0', '1', '2', '3', '4', '5', '6', '7', '8']
    mock_console_io = MockConsoleIO(values)
    console = Console(io=mock_console_io)
    runner = GameRunner(console=console)

    app.main(runner)

    assert mock_console_io.last_output == filled_board_output
```

Maybe you'll like it :)